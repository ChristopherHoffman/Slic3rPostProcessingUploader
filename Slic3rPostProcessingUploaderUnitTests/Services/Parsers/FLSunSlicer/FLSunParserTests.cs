using Slic3rPostProcessingUploader.Services.Parsers.FLSunSlicer;
using Snapshooter.MSTest;


namespace Slic3rPostProcessingUploaderUnitTests.Services.Parsers.FLSunSlicer
{
    [TestClass]
    public class FLSunParserTests
    {
        [TestMethod]
        public void ShouldReturnAnEmptySettingWhenGcodeIsEmpty()
        {
            var parser = new FLSunParser("");
            var result = parser.ParseGcode("");
            Assert.IsNotNull(result);

            Snapshot.Match(result);
        }

        [TestMethod]
        public void ShouldReturnExpectedValuesWhenGivenFullGcode()
        {
            var parser = new FLSunParser("");
            var result = parser.ParseGcode(FLSunParserTestGcode.CalibrationCube);

            Snapshot.Match(result);
        }

        [TestMethod]
        public void ShouldRenderTheExpectedNoteWhenGivenATemplateWithNoReplacements()
        {
            string template = "Settings:";

            var parser = new FLSunParser(template);
            var result = parser.ParseGcode(FLSunParserTestGcode.CalibrationCube);

            Assert.AreEqual("Settings:", result.settings.note);
        }

        [TestMethod]
        public void ShouldRenderTheExpectedNoteWhenGivenATemplateWithASingleReplacement()
        {
            string template = """
                Settings:
                    Layer Height: {{layer_height}}
                """;

            var parser = new FLSunParser(template);
            var result = parser.ParseGcode(FLSunParserTestGcode.CalibrationCube);

            Assert.AreEqual("""
                Settings:
                    Layer Height: 0.2
                """, result.settings.note);
        }

        [TestMethod]
        public void ShouldRenderTheExpectedNoteWhenGivenATemplateWithMultipleReplacements()
        {
            string template = """
                Settings:
                    Layer Height: {{layer_height}}
                    First Layer Height: {{first_layer_height}}
                    Wall Loops: {{wall_loops}}
                    Top Shell Layers: {{top_shell_layers}}
                    Bottom Shell Layers: {{bottom_shell_layers}}
                    Sparse Infill Density: {{sparse_infill_density}}
                """;

            var parser = new FLSunParser(template);
            var result = parser.ParseGcode(FLSunParserTestGcode.CalibrationCube);

            Assert.AreEqual("""
                Settings:
                    Layer Height: 0.2
                    First Layer Height: 0.300
                    Wall Loops: 3
                    Top Shell Layers: 4
                    Bottom Shell Layers: 3
                    Sparse Infill Density: 15%
                """, result.settings.note);
        }

    }
}
