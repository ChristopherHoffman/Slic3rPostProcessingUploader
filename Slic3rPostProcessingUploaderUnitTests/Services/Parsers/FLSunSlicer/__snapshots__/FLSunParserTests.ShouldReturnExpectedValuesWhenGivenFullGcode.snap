{
  "Slicer": "FLSunSlicer",
  "CuraVersion": "2.0.2",
  "PluginVersion": null,
  "settings": {
    "note": "Settings:\n\nQuality:\n  Layer Height:\n    Layer Height: 0.2\n    First Layer Height: 0.300\n  Line Width:\n    Default Line Width: 0.42\n  Seam:\n    Seam Position: aligned\n  Precision:\n    Precise Wall: 0\n  Ironing:\n    Ironing: no ironing\n  Wall Generator:\n    Generator: classic\n  Walls And Surfaces:\n    Only One Wall on Top Surfaces: 1\n\nStrength:\n  Walls:\n    Wall Loops: 3\n  Top/Bottom Shells:\n    Top Shell Layers: 4\n    Top Shell Thickness: \n    Bottom Shell Layers: 3\n    Bottom Shell Thickness: \n  Infill:\n    Sparse Infill Density: 15%\n    Sparse Infill Pattern: simplecrosshatch\n\nSpeed:\n  First Layer Speed:\n    First Layer: 60\n    Number of Slow Layers: 1\n  Other Layers Speed:\n    Outer Wall: 120\n    Inner Wall: 200\n    Sparse Infill: 500\n    Internal Solid Infill: 300\n    Top Surface: 300\n  Travel Speed:\n    Travel: 600\n\nSupport:\n  Support:\n    Enable Support: 0\n    Type: normal(auto)\n    Style: default\n    Threshold Angle: 30\n    On Build Plate Only: 0\n  Raft:\n    Raft Layers: 0\n\nMultimaterial:\n  Prime Tower:\n    Enable: 0\n    Width: 60\n    Brim Width: 3\n  Ooze Prevention:\n    Enable: 0\n  Flush Options:\n    Flush into Objects' Infill: 0\n    Flush into Objects' Support: 1\n  Advanced:\n    Use Beam Interlocking: \n\nOther:\n  Skirt:\n    Skirt Type: \n    Skirt Loops: 0\n    Skirt Min Extrusion Length: 0\n  Brim:\n    Brim Type: no_brim\n    Brim Width: 5\n  Special Mode:\n    Slicing Mode: regular\n    Print Sequence: by layer\n    Spiral Vase: 0\n    Fuzzy Skin: none",
    "print_name": null,
    "time_stamp": null,
    "estimated_print_time_seconds": 53389,
    "material_used_mg": 275720,
    "Snapshot": null,
    "file_name": null,
    "filamentUsage": null
  }
}
